logstash-core-event-java/src/test/java/com/logstash/KeyNodeTest.java

```diff
@@ -0,0 +1,27 @@
+package com.logstash;
+
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.junit.Assert.assertEquals;
+
+public class KeyNodeTest {
```

Comment:
value per se cannot be null because:

```java
   if (value != null) {
            if (value instanceof List) {
                return join((List)value, ",");
     
```

but that does not account for the List content, if it contains `null` then `join` would crash. I am not sure if it's possible for the List to contain nulls, @ph? otherwise it would be simple to add a `null` test in `join` to return `""` but I am not sure if that makes sense, specifically for the case where having `null`s in the List would indicate a bug in which case we could argue that crashing in `join` would be the good thing.